src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -430,8 +437,8 @@ partial class BoundLiteral : ILiteralExpression
 
     partial class BoundObjectCreationExpression : IObjectCreationExpression
     {
-        private static readonly ConditionalWeakTable<BoundObjectCreationExpression, object> s_memberInitializersMappings =
-            new ConditionalWeakTable<BoundObjectCreationExpression, object>();
+        private static readonly ConditionalWeakTable<BoundObjectCreationExpression, StrongBox<ImmutableArray<IMemberInitializer>>> s_memberInitializersMappings =
+            new ConditionalWeakTable<BoundObjectCreationExpression, StrongBox<ImmutableArray<IMemberInitializer>>>();
 
```

Comment:
I agree that in terms of memory allocation, there may not be much difference, but the downside is confusion: readers will wonder why ```StrongBox<T>``` was used.

If the goal is avoid casting, perhaps add a ```class RefWrapper<T> where T : struct { T Value; }``` and use that instead. But `object` would be simpler.